components/core/src/utils/make_dictionaries_readable/make-dictionaries-readable.cpp

```diff
@@ -81,8 +86,15 @@ int main (int argc, const char* argv[]) {
 
         file_writer.write_string(replace_characters("\n", "n", human_readable_value, true));
         file_writer.write_char('\n');
+        std::set<segment_id_t> segment_ids_set = entry.get_ids_of_segments_containing_entry();
```

Comment:
* Let's add a newline before for readability
* You can use `const auto&`
* How about `segment_ids` (since we typically don't specify the type in the variable's name in C++)?

Reply:
Gotcha. You should still use `const std::set<segment_id_t>&` then to avoid generating a copy.